Several test failures from running the tests in random order, although I expect several of these are also present in master.

```
____________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_sizeof _____________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 78, in test_sizeof
    assert ccode(sz) == 'sizeof(%s)' % typename
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
_________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_CommaOperator _________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 20, in test_CommaOperator
    assert ccode(expr) == '(++(x), 2*x)'
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
__________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_goto_Label ___________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 29, in test_goto_Label
    assert ccode(g) == 'goto early_exit'
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
____________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_struct _____________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 93, in test_struct
    assert ccode(s) == (
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
_____________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_union _____________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 108, in test_union
    assert ccode(u) == (
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
_________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_PreIncrement __________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 66, in test_PreIncrement
    assert ccode(p) == '++(x)'
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
_________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_PreDecrement __________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 54, in test_PreDecrement
    assert ccode(p) == '--(x)'
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
____________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_alignof ____________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 14, in test_alignof
    assert ccode(ax) == 'alignof(x)'
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
_________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_PostIncrement _________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 72, in test_PostIncrement
    assert ccode(p) == '(x)++'
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
_________________________________________________________ sympy/codegen/tests/test_cnodes.py:test_PostDecrement _________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cnodes.py", line 60, in test_PostDecrement
    assert ccode(p) == '(x)--'
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
_______________________________________________ sympy/codegen/tests/test_rewriting.py:test_create_expand_pow_optimization _______________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_rewriting.py", line 262, in test_create_expand_pow_optimization
    assert cc(x**4) == 'x*x*x*x'
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_rewriting.py", line 259, in <lambda>
    cc = lambda x: ccode(
TypeError: 'module' object is not callable
_________________________________________________________________________________________________________________________________________________________________________
____________________________________________________________ sympy/codegen/tests/test_cxxnodes.py:test_using ____________________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/codegen/tests/test_cxxnodes.py", line 11, in test_using
    assert cxxcode(u1) == 'using std::vector'
TypeError: 'module' object is not callable

_________________________________________________________________________________________________________________________________________________________________________
___________________________________________ sympy/multipledispatch/tests/test_core.py:test_inheritance_and_multiple_dispatch ____________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/multipledispatch/tests/test_core.py", line 78, in test_inheritance_and_multiple_dispatch
    assert f(A(), C()) == (A, C)
AssertionError
_________________________________________________________________________________________________________________________________________________________________________
_____________________________________________________ sympy/physics/units/tests/test_quantities.py:test_convert_to ______________________________________________________
Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/physics/units/tests/test_quantities.py", line 37, in test_convert_to
    assert q.convert_to(m) == 5000*m
AssertionError
```

--- a/sympy/codegen/tests/test_algorithms.py
+++ b/sympy/codegen/tests/test_algorithms.py
@@ -6,7 +6,7 @@ from sympy.codegen.fnodes import bind_C
 from sympy.codegen.futils import render_as_module as f_module
 from sympy.codegen.pyutils import render_as_module as py_module
 from sympy.external import import_module
-from sympy.printing import ccode
+from sympy.printing.codeprinter import ccode
 from sympy.utilities._compilation import compile_link_import_strings, has_c, has_fortran
 from sympy.utilities._compilation.util import may_xfail
 from sympy.testing.pytest import skip, raises

--- a/sympy/codegen/tests/test_applications.py
+++ b/sympy/codegen/tests/test_applications.py
@@ -3,7 +3,7 @@
 import tempfile
 
 from sympy.external import import_module
-from sympy.printing import ccode
+from sympy.printing.codeprinter import ccode
 from sympy.utilities._compilation import compile_link_import_strings, has_c
 from sympy.utilities._compilation.util import may_xfail
 from sympy.testing.pytest import skip

--- a/sympy/codegen/tests/test_cnodes.py
+++ b/sympy/codegen/tests/test_cnodes.py
@@ -1,5 +1,5 @@
 from sympy.core.symbol import symbols
-from sympy.printing import ccode
+from sympy.printing.codeprinter import ccode
 from sympy.codegen.ast import Declaration, Variable, float64, int64, String, CodeBlock
 from sympy.codegen.cnodes import (
     alignof, CommaOperator, goto, Label, PreDecrement, PostDecrement, PreIncrement, PostIncrement,

--- a/sympy/codegen/tests/test_cxxnodes.py
+++ b/sympy/codegen/tests/test_cxxnodes.py
@@ -1,7 +1,7 @@
 from sympy.core.symbol import Symbol
 from sympy.codegen.ast import Type
 from sympy.codegen.cxxnodes import using
-from sympy.printing import cxxcode
+from sympy.printing.codeprinter import cxxcode
 
 x = Symbol('x')
 

--- a/sympy/codegen/tests/test_fnodes.py
+++ b/sympy/codegen/tests/test_fnodes.py
@@ -13,7 +13,7 @@ from sympy.codegen.fnodes import (
 from sympy.codegen.futils import render_as_module
 from sympy.core.expr import unchanged
 from sympy.external import import_module
-from sympy.printing import fcode
+from sympy.printing.codeprinter import fcode
 from sympy.utilities._compilation import has_fortran, compile_run_strings, compile_link_import_strings
 from sympy.utilities._compilation.util import may_xfail
 from sympy.testing.pytest import skip, XFAIL

--- a/sympy/codegen/tests/test_rewriting.py
+++ b/sympy/codegen/tests/test_rewriting.py
@@ -2,7 +2,7 @@ import tempfile
 from sympy import log, exp, cos, S, Symbol, Pow, sin, MatrixSymbol, sinc, pi
 from sympy.assumptions import assuming, Q
 from sympy.external import import_module
-from sympy.printing import ccode
+from sympy.printing.codeprinter import ccode
 from sympy.codegen.matrix_nodes import MatrixSolve
 from sympy.codegen.cfunctions import log2, exp2, expm1, log1p
 from sympy.codegen.numpy_nodes import logaddexp, logaddexp2

--- a/sympy/core/add.py
+++ b/sympy/core/add.py
@@ -992,7 +992,7 @@ class Add(Expr, AssocOp):
             logflags = dict(deep=True, log=True, mul=False, power_exp=False,
                 power_base=False, multinomial=False, basic=False, force=False,
                 factor=False)
-            old = old.expand(logflags)
+            old = old.expand(**logflags)
         expr = expand_mul(old)
 
         if not expr.is_Add:

--- a/sympy/core/cache.py
+++ b/sympy/core/cache.py
@@ -71,7 +71,7 @@ def __cacheit(maxsize):
             try:
                 retval = cfunc(*args, **kwargs)
             except TypeError as e:
-                if not e.args[0].startswith('unhashable type:'):
+                if not e.args or not e.args[0].startswith('unhashable type:'):
                     raise
                 retval = func(*args, **kwargs)
             return retval

